src/tables/sortable-table.js

```diff
@@ -36,37 +36,89 @@ import classnames from 'classnames'
 /* eslint react/jsx-key: 0 */
 
 const propTypes = {
+  className: PropTypes.string,
   columns: PropTypes.arrayOf(Types.column).isRequired,
-  controlled: PropTypes.bool.isRequired,
   data: PropTypes.arrayOf(PropTypes.object).isRequired,
+  initialSortPath: PropTypes.string.isRequired,
+  initialSortFunc: PropTypes.func,
+  initialAscending: PropTypes.bool,
+  disableReverse: PropTypes.bool,
   disableSort: PropTypes.bool.isRequired,
+  controlled: PropTypes.bool,
+  onChange: PropTypes.func,
   rowComponent: Types.component,
   headerComponent: Types.component,
-  className: PropTypes.string,
-  ...sortablePropTypes,
 }
-
 const defaultProps = {
-  className: ''
+  className: '',
+  initialAscending: true,
+  disableReverse: false,
+  controlled: false,
+  onChange: noop,
 }
 
-function SortableTable ({
+
+function SortableTable({
+  className,
   columns,
   data: unsortedData,
+  initialSortPath,
+  initialSortFunc,
+  initialAscending,
+  disableReverse,
   disableSort,
   controlled,
-  sort,
-  ascending,
-  sortPath,
-  setSortPath,
-  setSortFunc,
+  onChange,
   rowComponent,
   headerComponent,
-  className,
 }) {
-  const data = (controlled || disableSort) ? unsortedData : sort(unsortedData)
+  const [sortControls, setSortControls] = useState({
```

Comment:
I'd suggest splitting this into three `useStates`. It might seem more verbose at first but it will save you some potential headaches, and be more performant too.

Reply:
Done, as discussed